webapp/packages/plugin-codemirror6/src/SearchPanel/SearchPanel.tsx

```diff
@@ -0,0 +1,249 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { closeSearchPanel, findNext, findPrevious, replaceAll, replaceNext, SearchQuery, setSearchQuery } from '@codemirror/search';
+
+import type { EditorView } from '@codemirror/view';
+import { useState } from 'react';
+
+import { Icon, useTranslate } from '@cloudbeaver/core-blocks';
+import { clsx, IconButton, Input } from '@dbeaver/ui-kit';
+
+import './SearchPanel.css';
+
+export function SearchPanel({
+  view,
```

Comment:
let's remove `view` from props, expose:
```
onQueryCahnge
onCaseSensetiveToggle
onLiteralToggle
onWholeWordToggle
onReplaceChange
onFindNext
onFindPrevious
onReplaceAll
onReplaceNext
onClose
```

The idea is to keep logic coupled with `view` in the same component / or hook. The coupled logic is split into two components and can't be used separately.